### PR TITLE
ci: cache jest

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes

Store jest's cache between GitHub actions runs.

- [First run](https://github.com/renovatebot/renovate/actions/runs/3198763199/attempts/1) (cache miss)
	- Node 14: 12m 54s
	- Node 16: 16m 8s
- [Second run](https://github.com/renovatebot/renovate/actions/runs/3198763199/attempts/2) (cache hit)
	- Node 14: 8m 8s
	- Node 16: 16m 33s
- [Third run](https://github.com/renovatebot/renovate/actions/runs/3198763199/attempts/3) (cache hit)
	- Node 14: 8m 10s
	- Node 16: 13m 43s

Worst case provides the same performance as a cache miss. Best case is 33% speed up.

## Context

#16203

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -83,6 +83,14 @@ jobs:
       - name: Build
         run: yarn build
 
+      - name: Cache jest cache
+        uses: actions/cache@56461b9eb0f8438fd15c7a9968e3c9ebb18ceff1 # tag=v3.0.10
+        with:
+          path: .cache/jest
+          key: ${{ runner.os }}-${{ matrix.node-version }}-jest-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ matrix.node-version }}-jest-
```
Comment (viceice) [inline .github/workflows/build.yml:92]: i think this will blow up the cache from time to time because jest isn't removing outdated cache files

Comment (JamieMagee) [reply .github/workflows/build.yml:92]: True, but I think having the hash of `yarn.lock` in there will likely be enough to prevent this.

Comment (viceice) [reply .github/workflows/build.yml]: no, the restore key will restore the old cache and so we get old files into new cache

Comment (JamieMagee) [reply .github/workflows/build.yml]: Okay, I'll remove the alternative restore-key.

**Comment by rarkins** [comment]
Are we designed to Node 16 being slow forever or are there things we or Jest can improve?

**Comment by viceice** [comment]
i don't think we can do much on renovate side. it's a known jest / node issue. 🤷‍♂️

**Review by rarkins — APPROVED** [review]

**Pull request merged**